### PR TITLE
fix: seatbelt sign sync to fsuipc

### LINKED ISSUE #2752: [REQUEST] Add FSUIPC support to seatbelt sign
**Is your feature request related to a problem? Please describe.**

Currently, the seatbelt switch does not trigger the seatbelt sign in FSUIPC (for SLC or PACX)

**Describe the feature you'd like to see implemented**

Add the trigger so it can tell FSUIPC if the seatbelt sign is on or not

**References**

N/A

**Additional context**

Discord username (if different from GitHub):

### PR DESCRIPTION
**Pull request by Saschl**
Fixes #2752

## Summary of Changes

Integrated the key event that is being sent by the default plane which allows fsuipc (and therefore SLC or PACX) to recognize that

## Screenshots (if necessary)

https://user-images.githubusercontent.com/19493808/111877824-601c2b00-89a5-11eb-800a-16537062f853.mp4

## Additional context

Discord username (if different from GitHub): saschl#9432

## Testing instructions

I don't know another way to test this besides having SLC or PACX.
1. Load in a flight and start the tool (SLC or PACX)
2. Switch the seatbelt signs button to on -> the tool should update accordingly
3. Switch the seatbelt signs button to off -> the tool should update accordingly
4. Perform 2. and 3. but toggle the seatbelt via the external tool. In game state should be updated accordingly

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page

### CODE REVIEW

**Review by rubenarruda — APPROVED** [review]

**Comment by lukecologne** [comment]
The problem with this approach is that the switch position could get out of sync if an external program also triggers the event, or sets the Simvar directly (I think the associated Simvar is `CABIN SEATBELTS ALERT SWITCH`, correct me if I'm wrong).

**Review by Benjozork — APPROVED** [review]
LGTM

**Comment by Saschl** [comment]
> The problem with this approach is that the switch position could get out of sync if an external program also triggers the event, or sets the Simvar directly (I think the Simvar associated with the event is `CABIN SEATBELTS ALERT SWITCH`, correct me if I'm wrong).

Hm you're right, but the switch position is controlled by `XMLVAR_SWITCH_OVHD_INTLT_SEATBELT_Position`. I just thought about syncing it in a <Update> tag but that would trigger the K event all the time afraid.

**Comment by lukecologne** [comment]
I mean the simple solution would be to just switch over to the sims own seatbelt simvar, which I assume the K event changes (i.e. `CABIN SEATBELTS ALERT SWITCH` as I commented above).

**Comment by FlightSimmer7700** [comment]
Category: QA Tester
Name: FlightSimmer7700#3468
Date of testing: 22/3/21
Version of the sim: 1.14.5.0
PR Tested: #4040 
Tier of testing: 1
Changes to observe:
Seatbelt sync through fsuipc

Testing technique: Loaded in and took a quick look
Steps you've made to test the Pull Request:
1. Loaded in
2. opened 0.7.8 FSUIPC
3. opened pacx
4 .started a flight
5. checked that seatbelt syncs

Media: Insert screenshots, recordings, or anything relevant to the review here.

Issues (If Any): None
Steps to achieve the issue

Overall Rating: Good
Conclusions of the testing: Syncs through as expected

**Comment by DavidEGTE** [comment]
Category: QA Tester
Name: David#2177
Date of testing: 23/03/2021
Version of the sim: 1.14.5.0
PR Tested: #4040
Tier of testing: 1

Changes to observe:
seatbelt sign sync to fsuipc

Testing technique: Try-out with SLC

Steps you've made to test the Pull Request:
1. Spawned C&D
2. Followed PR instructions 

Flight notes (When done a full flight):
N/A
Media: Insert screenshots, recordings, or anything relevant to the review here.
N/A
Issues (If Any): Write any issues
N/A
Steps to achieve the issue:
N/A

Overall Rating: Good
Conclusions of the testing: seatbelt signs switch to on/off syncs with SLC perfectly, GTG

**Comment by Watsi01** [comment]
Tested on a regular flight with FSUIPC (7.0.8) and PACX (latest version). 

Works perfect.

**Comment by Saschl** [comment]
Thanks all for the tests already! As Luke already mentioned we should also take care that the sync works in both directions. I already got it working, but the state from the flt files is not working currently. Will convert this to a draft until it's solved then we should be good :)

**Comment by stsvesuk** [comment]
Sorry, but for us poor old non slc/pacx/fsuipc developers (just simconnect api) it doesn't work. Please explain and yes, I may have missed/misunderstood in previous text, why the 'CABIN SEATBELTS ALERT SWITCH' simvar cannot be used.

**Comment by easyjetsimpilot** [comment]
Category: QA Tester
Name: Gavin (easyjetsimpilot)
Date of testing: 05/04/2021
Version of the sim: 1.14.5.0
PR Tested: #4040
Tier of testing: 1

Testing technique: Try-out with SLC
Steps you've made to test the Pull Request:

Test 1:
Spawned C&D - turned on the batteries and Ext power and confirmed the seatbelt sign config agrees with Self Loading Cargo and works perfectly - no Issues is spawning cold and dark with seatbelt sign in the 'off' position. 

Test 2:
Spawned ready to go - engines running and seat belt sign already in the 'on' position. In this case, SLC loads with the sign in the 'off' phase and so toggling the on/off switch in the sim has the reverse effect in SLC.
On (sim) = Off (slc)
Off (sim) = On (slc)

Overall Rating: Fair
Conclusions of the testing: Only seems to work correctly when the sim spawns cold and dark, which shouldn't affect most users who perform SLC flights. May need to be checked to see if on = on in all circumstances.

**Comment by Watsi01** [comment]
Should this be tested again or is it still WIP?

**Comment by Saschl** [comment]
I need to test it again, it's unfortunately a bit of a pain as reloading the plane in dev-tools != loading the plane from the world map so I need to load in from the world map for every change. WIll have a look this evening

File: flybywire-aircraft-a320-neo/ModelBehaviorDefs/A32NX/Airbus.xml [hunk 1]
```diff
@@ -764,6 +764,46 @@
...
+					</ANIM_CODE>
+				</UseTemplate>
+			</True>
+			<False>
+		<UseTemplate Name = "ASOBO_GT_Interaction_LeftSingle_Event">
+		</UseTemplate>
+			</False>
+		</Condition>
+		</Component>
+	</Template>
```
Comment (Benjozork) [inline flybywire-aircraft-a320-neo/ModelBehaviorDefs/A32NX/Airbus.xml:805]: Please format this correctly

File: flybywire-aircraft-a320-neo/SimObjects/AirPlanes/FlyByWire_A320_NEO/model/A320_NEO_INTERIOR.xml [hunk 2]
```diff
@@ -1217,18 +1217,20 @@
...
-                <UseTemplate Name="A32NX_GT_Switch_Dummy">
+                <Update Once="True">
+                    (L:XMLVAR_SWITCH_OVHD_INTLT_SEATBELT_Position) 1 == (A:CABIN SEATBELTS ALERT SWITCH, bool) 0 == and if{
+                      (&gt;K:CABIN_SEATBELTS_ALERT_SWITCH_TOGGLE)
+                    }
+                </Update>
+                <UseTemplate Name="FBW_GT_Switch">
                     <NODE_ID>SWITCH_OVHD_INTLT_SEATBELT</NODE_ID>
-                    <PART_ID>Seatbelt_Switch</PART_ID>
+					<ANIM_SIMVAR>CABIN SEATBELTS ALERT SWITCH</ANIM_SIMVAR>
```
Comment (Benjozork) [inline flybywire-aircraft-a320-neo/SimObjects/AirPlanes/FlyByWire_A320_NEO/model/A320_NEO_INTERIOR.xml:1230]: Please format this correctly

**Review by Benjozork — DISMISSED** [review]

File: flybywire-aircraft-a320-neo/SimObjects/AirPlanes/FlyByWire_A320_NEO/model/A320_NEO_INTERIOR.xml (hunk 2 above)
Comment (Saschl) [reply flybywire-aircraft-a320-neo/SimObjects/AirPlanes/FlyByWire_A320_NEO/model/A320_NEO_INTERIOR.xml:1255]: Strange, it appears fine in the IDE and the complete view from git. I reformatted it but it still looks strange here

Comment (Saschl) [reply flybywire-aircraft-a320-neo/SimObjects/AirPlanes/FlyByWire_A320_NEO/model/A320_NEO_INTERIOR.xml]: Alright replaced all tabs

**Comment by MisterChocker** [comment]
@Benjozork have your requests been resolved?

**Comment by Watsi01** [comment]
Hi all,
I really don't want to rush things here. And I want to ask objectively and in a really nice way: No bad feelings here. It seems there is kind of a reluctance to merge this. I just wanna know what the problem is and if it can be solved?

**Comment by ElonKerman** [comment]
Category: QA Tester
Name:  Elon Kerman#9253
Date of testing: 09/05/2021
Version of the sim: 1.15.10.0
PR Tested: #4040
Tier of testing: 1

Changes to observe:
seatbelt sign sync to fsuipc

Testing technique: Try-out with SLC

Steps you've made to test the Pull Request:

Spawned Cold and Dark
Followed PR instructions
Flight notes (When done a full flight):
N/A

Media: Insert screenshots, recordings, or anything relevant to the review here.
N/A

Issues (If Any): Write any issues
none

Steps to achieve the issue:
N/A

Overall Rating: Good
Conclusions of the testing: seatbelt signs switch to on/off syncs with SLC perfectly, LGTM

**Comment by Saschl** [comment]
@Watsi01 Yes you're right, technically we just need an approval here and we're good to go IMO.

**Comment by frankkopp** [comment]
Category: QA Tester
Name: Cdr_Maverick#6475
Date of testing: Current date 09/05/2021
Version of the sim: 1.15.10
PR Tested: #4040
Tier of testing: 1

Changes to observe:
Load in a flight and start the tool (SLC or PACX)
Switch the seatbelt signs button to on -> the tool should update accordingly
Switch the seatbelt signs button to off -> the tool should update accordingly
Perform 2. and 3. but toggle the seatbelt via the external tool. In game state should be updated accordingly

Testing technique: Tryout C&D at gate

Steps you've made to test the Pull Request:
1. C&D at gate - turned on cockpit and External Power
2. Connected Jetway and started boarding with SLC
3. Used UI OHVD Seatbelt switch, external switch (Logitech Panel, Stream Deck) to turn on/off seatbelts
4. Used SLC to turn on/off seatbelts

Issues (If Any): No issues - everything (UI OVHD, SCL, Stream Deck) stays in sync not matter how fast or in what order I switch

Overall Rating: Good
Conclusions of the testing: LGTM

**Comment by darkfury334** [comment]
Category: QA Trial
Name: DarkFury#4936
Date of testing: Current date 10/05/21
Version of the sim: 1.15.10.0
PR Tested: #4040
Tier of testing: 1
Changes to observe:
key event that is being sent by the default plane which allows fsuipc (and therefore SLC or PACX) to recognize that

Testing technique: Description of how you tested the Pull Request

Steps you've made to test the Pull Request: Test 
1. Spawned at EHAm C&D, loaded flight in SLC
2. Toggles the seatbelt signs button on/off in Aircraft  -> Checking tool had updated accordingly
3. Performed step 2 using SLC  -> Checking in game state had updated accordingly
4.
5.

Flight notes (When done a full flight):N/A

Media: 

Issues (If Any): N/A

Overall Rating: Good
Conclusions of the testing: Looks good. Working as expected

**Review by tracernz — APPROVED** [review]
Previously requested changes addressed and I can find no tabs. Code should ideally be in RNP rather than RPN but since this is already tested and ready to go I guess it's okay.

**Comment by airfirefighter** [comment]
> @Watsi01 Yes you're right, technically we just need an approval here and we're good to go IMO.

sorry for typing here ! i have the feeling this PR is cursed!! i just kidding!thanks all of you for ur grate work!!!

**Comment by Watsi01** [comment]
Finallly. Thanks to all!!

**Comment by airfirefighter** [comment]
> Finallly. Thanks to all!!
for me dosent work........  :(     i send a bug report....

**Comment by stsvesuk** [comment]
Works for me. I'm not using FSUIPC ... just a straight Simconnect call.

**Pull request merged**